scripts/method2.r

```diff
@@ -0,0 +1,139 @@
+## Method 2
+# Assignment 1:  
+library(tweedie) 
+library(ggplot2)
+library(parallel)
+library(doParallel)
+library(foreach)
+
+
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+
+
+# Assignment 2:  
+MTweedieTests <-  
+  function(N,M,sig){ 
+    sum(replicate(M,simTweedieTest(N)) < sig)/M 
+  } 
+
+
+# Assignment 3:  
+df <-  
+  expand.grid( 
+    N = c(10,100,1000,5000, 10000), 
+    M = 1000, 
+    share_reject = NA) 
+
+
+# Set up parallel processing using doParallel
+cl <- makeCluster(detectCores())
```

Comment:
I would perhaps use a few less cores than what you have, if you are using them for other things as well.